src/compiler/crystal/command.cr

```diff
@@ -511,9 +520,30 @@ class Crystal::Command
       @color = false
       compiler.color = false
     end
+    setup_compiler_warning_options(opts, compiler)
     opts.invalid_option { }
   end
 
+  private def setup_compiler_warning_options(opts, compiler)
+    compiler.warnings_exclude << Crystal.normalize_path "lib"
+    opts.on("--warnings emit|error|ignore", "How to treat warning. (default: emit)") do |w|
+      compiler.warnings = case w
+                          when "emit"
+                            Crystal::Warnings::Emit
+                          when "error"
+                            Crystal::Warnings::Error
+                          when "ignore"
+                            Crystal::Warnings::Ignore
+                          else
+                            error "--warnings should be emit, error, or ignore"
+                            raise "unreachable"
```

Comment:
It's strange that this line is needed.

Reply:
Do you mean the `raise "unreachable"`? 

The `error` function is not a `NoReturn` it depends on an argument value whether or not it does an `exit`.

Reply:
Aah... right 👍